sdk/python/v1beta1/kubeflow/katib/api/katib_client.py

```diff
@@ -415,7 +415,10 @@ class name in this argument.
             experiment.spec.max_failed_trial_count = max_failed_trial_count
 
         # If users choose to use a custom objective function.
-        if objective is not None:
+        if objective is not None or parameters is not None:
+            if objective is None or base_image is None or parameters is None:
+                raise ValueError("One of the required parameters is None")
+
```

Comment:
Why do you need this if you have the check here: https://github.com/kubeflow/katib/blob/e68fe38f5f994526146defc38afb66a6d79cd0aa/sdk/python/v1beta1/kubeflow/katib/api/katib_client.py#L366-L374

Reply:
Here, I added code to handle the case where users specify only "objective" or "parameters". This is not covered in the previous check.

Reply:
I see, similar to how we have check here: https://github.com/kubeflow/katib/blob/e68fe38f5f994526146defc38afb66a6d79cd0aa/sdk/python/v1beta1/kubeflow/katib/api/katib_client.py#L509-L513

Reply:
Yes, exactly.